netrd/distance/netsimile.py

```diff
@@ -102,37 +102,50 @@ def feature_extraction(G):
 
     # average degree of neighborhood
     neighbor_degs = [
-        np.mean([node_degree_dict[m]
-                 for m in egonets[n].nodes if m != n]) if node_degree_dict[n] > 0 else 0
+        np.mean([node_degree_dict[m] for m in egonets[n].nodes if m != n])
+        if node_degree_dict[n] > 0
+        else 0
         for n in node_list
     ]
 
     # average clustering coefficient of neighborhood
     neighbor_clusts = [
-        np.mean([node_clustering_dict[m]
-                 for m in egonets[n].nodes if m != n]) if node_degree_dict[n] > 0 else 0
+        np.mean([node_clustering_dict[m] for m in egonets[n].nodes if m != n])
+        if node_degree_dict[n] > 0
+        else 0
         for n in node_list
     ]
 
     # number of edges in the neighborhood
     neighbor_edges = [
-        egonets[n].number_of_edges()
-        if node_degree_dict[n] > 0 else 0 for n in node_list
+        egonets[n].number_of_edges() if node_degree_dict[n] > 0 else 0
+        for n in node_list
     ]
 
     # number of outgoing edges from the neighborhood
     # the sum of neighborhood degrees = 2*(internal edges) + external edges
-    #node_features[:,5] = node_features[:,0] * node_features[:,2] - 2*node_features[:,4]
-    neighbor_outgoing_edges = [len([edge for edge in
-                                    set.union(*[set(G.edges(j)) for j in egonets[i].nodes]) if not
-                                    egonets[i].has_edge(*edge)]) for i in node_list]
+    # node_features[:,5] = node_features[:,0] * node_features[:,2] - 2*node_features[:,4]
+    neighbor_outgoing_edges = [
+        len(
```

Comment:
not super hot on this one

Reply:
It's not pretty but it probably shouldn't be a double list comprehension anyway.